internal/client/service.go

```diff
@@ -220,20 +221,44 @@ type Service interface {
 	// User Info
 	GetUserInfo(ctx context.Context) (*UserInfo, error)
 
+	// API Gateway methods
+
+	// Notebook
+	ImportNotebookFromFile(ctx context.Context, fileName string, content []byte, useCaseID string) (*ImportNotebookResponse, error)
+	GetNotebook(ctx context.Context, id string) (*Notebook, error)
+	UpdateNotebook(ctx context.Context, id string, useCaseID string) (*Notebook, error)
+	DeleteNotebook(ctx context.Context, id string) error
+
 	// Add your service methods here
 }
 
 // Service for the DataRobot API.
 type ServiceImpl struct {
-	client *Client
+	client      *Client
+	apiGWClient *Client
+}
+
+// Helper function to calculate APIGWEndpoint from Endpoint.
+func calculateAPIGWEndpoint(endpoint string) string {
+	// Your logic to transform endpoint to APIGWEndpoint
+	// For example, if you're replacing "/api/v2" with "/api-gw":
+	return strings.Replace(endpoint, "/api/v2", "/api-gw", 1)
```

Comment:
CC @asuslov for awareness.

I think with a plan for Notebooks APIs (hopefully) going GA soon there will be a rewrite rule in place similar to how buzok/genai has.

I mention it since `/api-gw` may or may not become a legacy version of URL we _may(?)_ want to drop support for at some point.

Reply:
Thanks for the heads up! 👍 I cleaned up how this gets built anyway because I needed the base URL to provide a handy link to the Notebook in my latest update. It should make it quite easy to adjust and/or drop from the TF provider.